api_server/src/http_service.rs

```diff
@@ -524,8 +524,16 @@ impl hyper::server::Service for ApiServerHttpService {
                             .lock()
                             .expect("Failed to acquire lock on MMDS info");
                         if mmds.is_initialized() {
-                            mmds.patch_data(json_value);
-                            Either::A(future::ok(empty_response(StatusCode::NoContent)))
+                            if mmds.patch_data(json_value).is_ok() {
+                                Either::A(future::ok(empty_response(StatusCode::NoContent)))
+                            } else {
+                                Either::A(future::ok(json_response(
+                                    StatusCode::BadRequest,
+                                    json_fault_message(
+                                        "Cannot add non-strings values to the MMDS data-store.",
```

Comment:
Instead of printing a message here, I suggest you implement to_string for `Error`. The `UnsupportedValueType` can have the message that you added here.

Reply:
In cases like this, when you need to convert a `Result` to another `Result`, the idiomatic Rust way would be to transform one into the other with `map`. Adding the `to_string` suggested by Andreea, it could look something like:

```rust
mmds.patch_data(json_value)
    .map(|| {
        Either::A(future::ok(empty_response(StatusCode::NoContent)))
    })
    .map_err(|e| {
        Either::A(future::ok(json_response(
            StatusCode::BadRequest,
            json_fault_message(e.to_string()),
        )))
    })
```

Reply:
@aghecenco Hi! Since we need to unwrap the `Result` for both `Ok()` and `Err()` cases, I have made the change to use `match`. Please let me know if there are better ways to do this! Thanks!

Reply:
I think any further Rust-ifications would make the code harder to read, so I believe the current form is best :+1: